Feature/integration test config

kkovarik opened the pull request:
Changed configuration to enable integration tests, see: https://github.com/OpenWiseSolutions/openhub-ri/pull/1

### CHANGES
* MvcPageableConfiguration removed from spring.factories. Now, when openhub-web module is referenced, this file is picked up and therefore beans from packages like org.openhubframework.openhub.admin are needed or spring context will not start. Switched to "normal" Configuration class.
* ~~added dependency to tomcat to test module, for integration tests to have embedded servlet container ready.~~
* ~~added TestWebConfig class with prepared autoscan for openhub-web beans~~

pjuza commented on test/src/main/java/org/openhubframework/openhub/test/TestWebConfig.java:
```diff
@@ -0,0 +1,34 @@
+ * @since 2.0
+ */
+@ComponentScan(basePackages = {
+        "org.openhubframework.openhub.web",
```
Previously was org.openhubframework.openhub.web.common, isn't it misspelling?

kkovarik replied:
No. There is more settings that I need for integration tests - like security, that is configured in another subpackage under org.openhubframework.openhub.web.config...so I added the parent...

hanusto replied:
Question is whether we should support web-based integration tests via `openhub-test`. Last result is for now NO. So it is correct to define `TestWebConfig` in openhub-test? Maybe compromise is moved this class to RI project where standard project specific classes will be created via archetype.

hanusto replied:
BTW: test support dependency for internal use, core of framework, is something other than support for testing project implementation. In tests to verify project implementation user probably wants to use dependency to core framework functions, but it is cyclic dependency, where test artifact will have dependency to core (to use classes) and core has dependency to test for testing support.

pjuza commented on test/pom.xml:
```diff
@@ -75,6 +75,11 @@
             <artifactId>spring-data-commons</artifactId>
         </dependency>
+        <!-- added tomcat to enable web tests, exclude when needed -->
+        <dependency>
```
Is it necessary? I wanted to keep test module as lighweight as possible, but if has to be ...

kkovarik replied:
No, it is only necessary for integration tests, will remove...

hanusto replied:
Remove

hanusto commented:
It is necessary to resolve conflicts.

kkovarik committed:
OHFJIRA-22 : MvcPageableConfiguration removed from spring.factories, initialized by Configuration annotation

- openhub-web module can be now referenced without having to have packages like org.openhubframework.openhub.admin on autoscan

kkovarik commented:
@hanusto @pjuza Rebased on top of current develop.

Changes from previous iteration:
- removed dependency to tomcat & TestWebConfig class as it will be solved in openhub-ri
